### PR TITLE
Added preview for networks for GCP PSC

### PR DESCRIPTION
**Pull request by jessiedu-db**

### CODE REVIEW

File: mws/resource_mws_networks_test.go [hunk 1]
```diff
@@ -107,6 +107,68 @@ func TestResourceNetworkCreate_GCP(t *testing.T) {
...
+		account_id = "abc"
+		network_name = "Open Workers"
+		gcp_network_info {
+			network_project_id = "project_a"
+			vpc_id = "vpc_a"
+			subnet_id = "subnet_a"
+			subnet_region = "region_a"
+			pod_ip_range_name = "pods"
+			service_ip_range_name = "svc"
+        }
```
Comment (grayluck) [inline mws/resource_mws_networks_test.go:160]: nit: fix the indentation?

**Review by grayluck — APPROVED** [review]

File: mws/resource_mws_networks_test.go (hunk 1 above)
Comment (jessiedu-db) [reply mws/resource_mws_networks_test.go:160]: Seems the indentation just looks weird on Github. Locally it looks fine. Indentation looks same with other tests.

Comment (grayluck) [reply mws/resource_mws_networks_test.go:160]: L160-164 uses spaces for indentation while the rest uses tabs. Can you check?

File: internal/acceptance/mws_networks_test.go [hunk 2]
```diff
@@ -22,3 +22,25 @@ func TestMwsAccNetworks(t *testing.T) {
...
+			gcp_network_info {
+			  network_project_id = "{env.GOOGLE_PROJECT}"
+			  vpc_id = "env.VPC_NETWORK_ID"
+			  subnet_id = "env.SUBNET_ID"
+			  subnet_region = "{env.GOOGLE_REGION}"
+			  pod_ip_range_name = "pod"
+			  service_ip_range_name = "svc"
+            }
+            vpc_endpoints {
+        	  rest_api = ["{env.REST_API_VPCE_ID}"]
```
Comment (nfx) [inline internal/acceptance/mws_networks_test.go:41]: These environment variables don't exist in our test infrastructure. I'll show you how to add those on Monday

**Review by nfx — CHANGES_REQUESTED** [review]

File: docs/resources/mws_networks.md [hunk 3]
```diff
@@ -161,6 +161,28 @@ resource "databricks_mws_networks" "this" {
...
+In order to create a VPC [that leverages GCP Private Service Connect](https://docs.gcp.databricks.com/administration-guide/cloud-configurations/gcp/private-service-connect.html) you would need to add the `vpc_endpoint_id` Attributes from [mws_vpc_endpoint](mws_vpc_endpoint.md) resources into the [databricks_mws_networks](databricks_mws_networks.md) resource. For example:
+
+```hcl
+resource "databricks_mws_networks" "this" {
+  account_id   = var.databricks_account_id
+  network_name = "test-demo-${random_string.suffix.result}"
+  gcp_network_info {
+    network_project_id    = var.google_project
+    vpc_id                = google_compute_network.dbx_private_vpc.name
+    subnet_id             = google_compute_subnetwork.network-with-private-secondary-ip-ranges.name
```
[the remark this answers is not in the document]
Comment (jessiedu-db) [reply docs/resources/mws_networks.md]: Done.

File: mws/resource_mws_networks_test.go (hunk 1 above)
Comment (jessiedu-db) [reply mws/resource_mws_networks_test.go]: Done.

File: docs/resources/mws_networks.md [hunk 4]
```diff
@@ -161,6 +161,28 @@ resource "databricks_mws_networks" "this" {
...
+    subnet_region         = google_compute_subnetwork.network-with-private-secondary-ip-ranges.region
+    pod_ip_range_name     = "pods"
+    service_ip_range_name = "svc"
+  }
+  vpc_endpoints {
+    dataplane_relay = [databricks_mws_vpc_endpoint.relay.vpc_endpoint_id]
+    rest_api        = [databricks_mws_vpc_endpoint.workspace.vpc_endpoint_id]
+  }
+  depends_on = [vpc_endpoint.workspace, vpc_endpoint.relay]
```
[the remark this answers is not in the document]
Comment (jessiedu-db) [reply docs/resources/mws_networks.md]: It should be fine after https://github.com/databricks/terraform-provider-databricks/pull/2080 is merged?

File: docs/resources/mws_networks.md [hunk 5]
```diff
@@ -98,7 +98,7 @@ resource "databricks_mws_networks" "this" {
     dataplane_relay = [databricks_mws_vpc_endpoint.relay.vpc_endpoint_id]
     rest_api        = [databricks_mws_vpc_endpoint.workspace.vpc_endpoint_id]
   }
-  depends_on = [aws_vpc_endpoint.workspace, aws_vpc_endpoint.relay]
+  depends_on = [vpc_endpoint.workspace, vpc_endpoint.relay]
```
[the remark this answers is not in the document]
Comment (jessiedu-db) [reply docs/resources/mws_networks.md]: Reverted back.

File: docs/resources/mws_networks.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (jessiedu-db) [reply docs/resources/mws_networks.md]: Removed this "depends_on"

**Review by jessiedu-db — COMMENTED** [review]
Dummy

**Review by nfx — APPROVED** [review]

**Pull request merged**